### PR TITLE
feat(sdk): Adding parameter to image to specify file type jpg, png, bmp, gif

### PR DESCRIPTION
**Pull request by fdsig**
Fixes
-----
- Fixes WB-14720

Description
-----------
Pillow has the functionality to save jpg, png, bmp, gif images and these can be uploaded to wandb -- we currently do this when adding a file path to an image but 

This feature request aligns what we upload when loading an image from the path to what is possible to save an image as from data -- was requested as a means to reduce the size of images uploaded -- jpg,bmp, gif are smaller files than .png.

### <samp>🤖 Generated by Copilot at 7e272ab</samp>

This pull request enhances the `Image` data type to handle different image file types. It adds a new parameter and attribute to store the file type, and updates the validation and saving logic accordingly. It also adds and fixes unit tests for the `Image` and `_ImageFileType` classes in the `test_dtypes.py` file.

Testing
-------
How was this PR tested?

Unit test update test_dtypes.py

This commit was also tested with:

```python
from PIL import Image
import numpy as np
import wandb

# Create a dummy 3-channel image (e.g., 256x256 RGB)
width, height = 256, 256
dummy_array = np.random.randint(0, 256, (height, width, 3), dtype=np.uint8)
dummy_image = Image.fromarray(dummy_array)

# List of formats to save in
formats = ["BMP", "JPG", "PNG","GIF"]
run = wandb.init(project="image-formats", job_type="save",entity="demonstrations")


# Save the image in each format
for file_format in formats:
    class_labels = {1: "tree", 2: "car", 3: "road"}
    dummy_image.save(f"dummy_image.{file_format.lower()}")
    #print(f"Saved in {file_format} format.")
    dummy_image = Image.fromarray(dummy_array)
    img = wandb.Image(
    dummy_image,
    boxes={
        "box_predictions": {
            "box_data": [
                {
                    "position": {
                        "minX": 0.1,
                        "maxX": 0.2,
                        "minY": 0.3,
                        "maxY": 0.4,
                    },
                    "class_id": 1,
                    "box_caption": "minMax(pixel)",
                    "scores": {"acc": 0.1, "loss": 1.2},
                },
            ],
            "class_labels": class_labels,
        },
        "box_ground_truth": {
            "box_data": [
                {
                    "position": {
                        "minX": 0.1,
                        "maxX": 0.2,
                        "minY": 0.3,
                        "maxY": 0.4,
                    },
                    "class_id": 1,
                    "box_caption": "minMax(pixel)",
                    "scores": {"acc": 0.1, "loss": 1.2},
                },
            ],
            "class_labels": class_labels,
        },
    },
    masks={
        "mask_predictions": {
            "mask_data": np.random.randint(0, 4, size=(30, 30)),
            "class_labels": class_labels,
        },
        "mask_ground_truth": {"path": f"dummy_image.{file_format.lower()}", "class_labels": class_labels},
    },file_type=file_format.lower()
)
    run.log({f"dummy_image.{file_format.lower()}": img})
run.finish()

```

workspace [here](https://wandb.ai/demonstrations/image-formats/runs/oc4kfcom)

Checklist
-------
- [x] Include reference to internal ticket "Fixes WB-NNNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at 7e272ab</samp>

> _`Image` class grows_
> _Supports more file types now_
> _Tests catch the bugs_

### CODE REVIEW

**Comment by dmitryduev** [comment]
hey @fdsig, any progress on this one?

**Comment by fdsig** [comment]
@dmitryduev This should be good to go now -- I needed to update the tests, reformat some code in unit tests, and remove a superfluous assignment of self.format in image.py that I had added necessarily when reasoning out the best palace to implement adding .jpg functionality.

**Comment by fdsig** [comment]
>>  pre-commit failing -- I have made multiple adjustments to nested dictionary to get this to pass -- however, I think that this is autocorrected -- but happy to revisit
>>  unit-linux-py37 fails [here](https://github.com/wandb/wandb/blob/453502bc170b6a680d2d295cf62cf55c5ca3c017/tests/pytest_tests/unit_tests/test_lib/test_redir.py#L173) however I am pretty sure this has no bearing on this pr
>>  codecov/project — 68.11% (-11.83%) compared to 99c2a2b I am unsure how this is measure -- however I tried to ensure that coverage has not reduced but happy to revisit/take guidance on what additional tests could be written here. 

The test should be:

- [x] --> Cover the case of file types updated image format 
- [x] --> Case of each file type and all anotations
- [x] --> case of file type as none

File: .vscode/settings.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   "python.linting.pylintEnabled": false,
   "jupyter.jupyterServerType": "local",
   "[python]": {
-    "editor.defaultFormatter": "charliermarsh.ruff",
+    "editor.defaultFormatter": "ms-python.autopep8",
```
Comment (dmitryduev) [inline .vscode/settings.json:32]: ruff's better ;)
```suggestion
    "editor.defaultFormatter": "charliermarsh.ruff",
```

**Review by dmitryduev — APPROVED** [review]
Beautiful, thanks a lot @fdsig!

**Pull request merged**